lib/internal/tls.js

```diff
@@ -23,6 +25,15 @@ function parseCertString(s) {
   return out;
 }
 
+function setupRootCertsFile() {
+  if (process.env.NODE_EXTRA_CA_CERTS) {
```

Comment:
One suggestion: I think this can be called unconditionally. certLoadExtraRootCertsFile is a null op if `extra_root_certs_file` hasn't been set, and it is only set if this env var is present, so this is a non-obvious optimization. I think its a bit fragile, if a CLI flag was introduced to set the `extra_root_certs_file`, for example, then this code would not execute, and the bug you are fixing would resurface for that use case.

Reply:
Another potential footgun: the C++ code does suid root checks (calls `SafeGetenv("NODE_EXTRA_CA_CERTS")`); `process.env` does not.

Reply:
I have removed the conditional statement as @sam-github suggested. After the removing, we don't use `process.env.NODE_EXTRA_CA_CERTS` in js land which avoids the potential footgun @bnoordhuis described.